test-d/foundry/common/utils/helpers.mjs.test-d.ts

```diff
@@ -202,6 +202,44 @@ declare class ClassWithConstructorParameters {
 expectType<boolean>(foundry.utils.isSubclass(ClassWithNoConstructorParameters, ClassWithConstructorParameters));
 expectType<boolean>(foundry.utils.isSubclass(ClassWithConstructorParameters, ClassWithNoConstructorParameters));
 
+// expandObject
+// hard to infer back
+expectType<{}>(foundry.utils.expandObject({}));
+expectType<{ test: number }>(foundry.utils.expandObject({ test: 1 }));
+expectType<{ test: number; deep: { value: number } }>(
+  foundry.utils.expandObject<{ test: number; deep: { value: number } }>({
+    test: 1,
+    "deep.value": 1
+  })
+);
+expectType<{ "0": number }>(foundry.utils.expandObject<{ "0": number }>({ "0": 1 }));
+
+// hasProperty
+expectType<boolean>(foundry.utils.hasProperty({ k1: 1 }, "k1"));
```

Comment:
I would just note that `hasProperty` is fundamentally not as ideal as it could in Typescript. It can't really be expressed as a type guard without running into a million unions.